dmoj/cptbox/isolate.py

```diff
@@ -314,63 +348,97 @@ def _file_access_check(
         # This works, except when the child process uses /proc/self, which refers to something else in this process.
         # Therefore, we "project" it by changing it to /proc/[tid] for computing the realpath and doing the samefile
         # check. However, we still keep it as /proc/self when checking access rules.
+
+        # normpath doesn't strip leading slashes
         projected = normalized = '/' + os.path.normpath(file).lstrip('/')
+        proc_dir = f'/proc/{debugger.tid}'
         if normalized.startswith('/proc/self'):
-            file = os.path.join(f'/proc/{debugger.tid}', os.path.relpath(file, '/proc/self'))
-            projected = '/' + os.path.normpath(file).lstrip('/')
-        elif normalized.startswith(f'/proc/{debugger.tid}/'):
+            projected = os.path.join(proc_dir, os.path.relpath(projected, '/proc/self'))
+        elif normalized.startswith(proc_dir):
             # If the child process uses /proc/getpid()/foo, set the normalized path to be /proc/self/foo.
             # Access rules can more easily check /proc/self.
-            normalized = os.path.join('/proc/self', os.path.relpath(file, f'/proc/{debugger.tid}'))
-        real = os.path.realpath(file)
+            normalized = os.path.join('/proc/self', os.path.relpath(normalized, proc_dir))
 
+        real = os.path.realpath(projected)
         try:
-            same = normalized == real or os.path.samefile(projected, real)
+            same = projected == real or os.path.samefile(projected, real)
```

Comment:
Was this change intended? It basically means the `elif normalized.startswith(proc_dir):` branch above is dead code, because the resulting `normalized` path is never used.

Reply:
It is more sensible to check `projected` with `normalized`. This way, we don't waste time if a process uses a `/proc` file, like `/proc/self/maps`, which we allow. As for `normalized`, it is still used for the jail access check, so the code is not dead. This change was intended.

Reply:
In the spirit of not doing too much in one PR, I'll revert this.